host/containerinit/init.go

```diff
@@ -412,20 +413,32 @@ func monitor(port host.Port, container *ContainerInit, env map[string]string) (d
 			}
 		}
 	}
-	addr := fmt.Sprintf("%s:%v", env["EXTERNAL_IP"], port.Port)
+	inst := &discoverd.Instance{
+		Addr:  fmt.Sprintf("%s:%v", env["EXTERNAL_IP"], port.Port),
+		Proto: port.Proto,
+	}
+	// add any Flynn environment variables to the meta
+	for k, v := range env {
+		if strings.HasPrefix(k, "FLYNN_") {
```

Comment:
I think we probably need a whitelist, this will pick up user-define variables as well as some other stuff that is not relevant like `FLYNN_POSTGRES`.